src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
+        var mat = List.turnIntoCSList([
+            List.turnIntoCSList([cen.value[2], zer, CSNumber.neg(cen.value[0])]),
+            List.turnIntoCSList([zer, cen.value[2], CSNumber.neg(cen.value[1])]),
+            List.turnIntoCSList([zer, zer, cen.value[2]])
+        ]);
+        var aa = List.normalizeZ(General.mult(mat, a));
+        var bb = List.normalizeZ(General.mult(mat, b));
+        var cc = List.normalizeZ(General.mult(mat, c));
+
+
+        // get angles of A and C 
+        var startAngle = -Math.atan2(aa.value[1].value.real, aa.value[0].value.real);
+        var endAngle = -Math.atan2(cc.value[1].value.real, cc.value[0].value.real);
+
+        cen = List.normalizeZ(cen);
+        var arcDist = List.abs(List.sub(a, cen));
+
+        // x, y vals of the center
+        var pt = [cen.value[0].value.real, cen.value[1].value.real];
+
+        // transform to canvas
+        var m = csport.drawingstate.matrix;
+        var xx = pt[0] * m.a - pt[1] * m.b + m.tx;
+        var yy = pt[0] * m.c - pt[1] * m.d - m.ty;
+
+
+        // check for counter clockwise drawing
+        var cclock = List.det3(a, b, c).value.real > 0;
+
+        // modifs handling
+        Render2D.handleModifs(modifs, Render2D.conicModifs);
+        var size = 4 / 2.5;
+        if (Render2D.size !== null)
+            size = Render2D.size;
+
+        csctx.save();
+        csctx.lineWidth = size;
+        csctx.strokeStyle = Render2D.lineColor;
+
+        // canvas circle radius 
+        var rad = arcDist.value.real * m.sdet;
+
+        csctx.beginPath();
+        csctx.translate(xx, yy);
+
+        // use the canvas arc function -- buggy in Chrome at least in Okt 15
+        var useArc = false;
+
+        if (useArc) {
+            csctx.arc(0, 0, arcDist.value.real * m.sdet, startAngle, endAngle, cclock);
+        } else {
+            var num = 500; // Number of segments
+
+            //  mod 2 pi in case startAngle > endAngle
+            if (startAngle > endAngle) endAngle = endAngle + Math.PI * 2;
+
+            // divide segments --  rotate counterclockwise if necessary
+            var ntler = !cclock ? (endAngle - startAngle) / num : -(2 * Math.PI - endAngle + startAngle) / num;
+
+            // drawing
+            csctx.moveTo(rad * Math.cos(startAngle), rad * Math.sin(startAngle));
+            var angl;
+            for (var ii = 0; ii <= num; ii++) {
+                angl = startAngle + ii * ntler;
+                csctx.lineTo(rad * Math.cos(angl), rad * Math.sin(angl));
+            }
+        }
+
+
+        if (df === "F" || false) {
+            csctx.fillStyle = Render2D.lineColor;
+            csctx.closePath();
+            csctx.fill();
+        }
+
+        if (df === "D") {
+            csctx.stroke();
+        }
+        csctx.restore();
+
+    } else { // segment case
+        var ptA = eval_helper.extractPoint(v1);
+        var ptB = eval_helper.extractPoint(v2);
+        var ptC = eval_helper.extractPoint(v3);
```

Comment:
`extractPoint` should be followed by checks against `ptA.ok` and the likes. Just return early if one of them is not `ok`.